packages/sdk-snippets/src/targets/csharp/dotnet6/webhooks/client.ts

```diff
@@ -10,116 +9,116 @@ export const dotnet6: Client = {
     link: 'https://docs.microsoft.com/en-us/dotnet/core/introduction',
     description: 'ReadMe Metrics Webhooks SDK usage on .NET 6.0',
   },
-  convert: ({ secret, security, server }, options) => {
-    const opts = {
-      indent: '  ',
-      ...options,
-    };
-
-    const { blank, join, push, pushVariable, ranges } = new CodeBuilder({ indent: opts.indent });
-
-    push('var builder = WebApplication.CreateBuilder(args);');
-    push('var app = builder.Build();');
-
-    blank();
-
-    push('// Your ReadMe secret');
-    push(`var secret = "${secret}";`);
-
-    blank();
-
-    push('app.MapPost("/webhook", async context =>');
-    push('{');
-    push('// Verify the request is legitimate and came from ReadMe.', 1);
-    push('var signature = context.Request.Headers["readme-signature"];', 1);
-    push('var body = await new StreamReader(context.Request.Body).ReadToEndAsync();', 1);
-
-    blank();
-
-    push('try', 1);
-    push('{', 1);
-    push('ReadMe.Webhook.Verify(body, signature, secret);', 2);
-    push('}', 1);
-    push('catch (Exception e)', 1);
-    push('{', 1);
-    push('context.Response.StatusCode = StatusCodes.Status401Unauthorized;', 2);
-    push('await context.Response.WriteAsJsonAsync(new', 2);
-    push('{', 2);
-    push('error = e.Message,', 3);
-    push('});', 2);
-    push('return;', 2);
-    push('}', 1);
-    blank();
-
-    push('// Fetch the user from the database and return their data for use with OpenAPI variables.', 1);
-    push('// @todo Write your own query logic to fetch a user by `body["email"]`.', 1);
-    blank();
-
-    push('await context.Response.WriteAsJsonAsync(new', 1);
-    push('{', 1);
-
-    if (!server.length && !security.length) {
-      push('// Add custom data to return in your webhook call here.', 2);
-    }
-
-    if (server.length) {
-      push('// OAS Server variables', 2);
-      server.forEach(data => {
-        pushVariable(
-          `${escapeForObjectKey(data.name, true)} = "${escapeForDoubleQuotes(
-            data.default || data.default === '' ? data.default : data.name
-          )}",`,
-          {
-            type: 'server',
-            name: data.name,
-            indentationLevel: 2,
+  convert: ({ secret, security, server }) => {
+    const writer = new CodeWriter({
+      indentNumberOfSpaces: 2,
+      useSingleQuote: false,
+    });
+
+    writer.writeLine('var builder = WebApplication.CreateBuilder(args);');
```

Comment:
There's some more boilerplate now but I feel like this is all easier to maintain. Maybe it's just me though.

Reply:
Damn, the library doesn't work when destructured:

![Screen Shot 2022-08-24 at 9 34 55 AM](https://user-images.githubusercontent.com/33762/186473909-c50ad200-c86a-4e3b-a005-a554059007d9.png)

I'm going to leave this as-is for now.

Reply:
Yeah weird, I wonder if it's some binding thing or something.